### ISSUE TITLE
Ranger freezes forever with "loading..." message

### ISSUE THREAD

**Issue by MaxGyver83** [ranger/ranger#2022]
#### Runtime Environment

- Operating system and version: Ubuntu 18.04
- Terminal emulator and version: GNOME Terminal 3.28.2 using VTE 0.52.2 +GNUTLS -PCRE2
- Python version: 3.6.9
- Ranger version/commit: ranger 1.9.3
- Locale: en_US.UTF-8

#### Current Behavior
When starting ranger, it just prints

```
loading...kranger\kranger\
```

forever. It can't even be cancelled with `Ctrl-C`.

#### Expected Behavior
Ranger starts instantly.

#### Context
Ranger startup.

#### Possible Solutions
No idea.

#### Steps to reproduce
1. pip3 install ranger-fm --user
2. ranger

#### Traceback
No traceback. But I set a logging file. That's the output (well, not much information):
```
10:37:46 INFO ranger version: ranger 1.9.3
10:37:46 INFO Python version: 3.6.9 (default, Apr 18 2020, 01:56:04) [GCC 8.4.0]
10:37:46 INFO Locale: en_US.UTF-8
10:37:46 INFO Process ID: 26643
```

Similar to https://github.com/ranger/ranger/issues/1883, but in my case `ranger` freezes forever, even without `screen`.

#### Debugging

I tried debugging that issue (`pudb3 ~/.local/bin/ranger`):

`ranger` freezes after entering `self.enter_dir(self.thistab.path)` in the function `loop` in `fm.py`. `self.thistab.path` is set to my home directory.

Next I tried to go deeper. This time I stepped through the function `self.enter_dir`but I could step through it without a freeze. I finally could see `ranger` opening with dirs and files. Maybe `ranger` needs that delay caused by using the debugger?!

**Comment by MaxGyver83** [comment]
Strange: After starting `ranger` in the debugger some times, now it works as expected both in gnome-terminal and alacritty.

**Comment by MaxGyver83** [comment]
After rebooting my laptop this issue occurs again.

**Comment by MaxGyver83** [comment]
At least I could go on debugging. `ranger` freezes in `assign_cursor_positions_for_subdirs(self)` in `tab.py`. Inside this function is a loop. The variable `last_path` has the value `'/lhome'` and `path` has the value `'/'` before freezing at the function call `path.move_to_obj(last_path)`.

After digging deeper, I think there might be an endless loop at this point:

![image](https://user-images.githubusercontent.com/13970628/86126778-2c2f6b80-badf-11ea-9311-de770be4de5c.png)

**Comment by toonn** [comment]
I don't see how it can get stuck without throwing an error. There's no infinite loops or recursion I can see.

**Comment by MaxGyver83** [comment]
I don't know what else I can do to find the cause for `ranger` loading forever. I have tested your commit but it still shows `loading...` forever.

**Comment by toonn** [comment]
I erroneously tagged that commit. Like I said, I have no idea what's going on here. Have you tried running `ranger -c`?

**Comment by MaxGyver83** [comment]
Same behaviour with `ranger -c`. When I'm in that for loop you can see in my screenshot, I press `n` many times until suddenly the debugger shows the console and never returns.

**Comment by MaxGyver83** [comment]
Do you know what this code actually does?

```python
for _ in self.load_generator:
    pass
self.load_generator = None
```

Should I search the problem in `load_bit_by_bit()` because of

```python
self.load_generator = self.load_bit_by_bit()
```

**Comment by MaxGyver83** [comment]
I just noticed that a simple `ls /` freezes. Probably this causes the freeze of `ranger` somehow.

**Comment by toonn** [comment]
Oh, definitely. Because ranger's trying to set the file selected in the / directory, because you might navigate upwards. So it has to list the / directory. Is this a permission issue?

**Comment by MaxGyver83** [comment]
All files and folders in `/` belong to root (checked with `sudo ls -l /`). But `ls` should always work, right?

**Comment by MaxGyver83** [comment]
The culprit is a symbolic link `/mapr`. If I skip this entry with `ll / -I mapr`, I get an output.

I don't know if such a case should be taken care of because it might be seldom. But on the other hand, it would be nice if I could use `ranger` in my home directory, no matter what's in `/`.

**Comment by toonn** [comment]
Why exactly does this symlink pose a problem?

**Comment by MaxGyver83** [comment]
The target of the symlink is a mount point for a network storage. And this seems to be "dead". It can't even be unmounted.

**Comment by MaxGyver83** [comment]
OMG: That's stupid. `ls` is an alias for `ls --color=auto`:

```shell
~ $ type ls
ls is aliased to `ls --color=auto'
```

`\ls /` works, `\ls --color=auto /` fails!

Is `ranger`affected by bash aliases?

**Comment by MaxGyver83** [comment]
BTW: It's a known issue for `ls`: https://bugs.launchpad.net/ubuntu/+source/coreutils/+bug/264313

**Comment by toonn** [comment]
Nope, but ranger *does* try to stat every file in every directory up to /. That's where it's getting stuck. No quick fix. #1540 touches on a somewhat related issue and might be a part of a fix.

**Comment by MaxGyver83** [comment]
In Linux I can combine `stat` with `timeout`:

```sh
timeout 0.5 stat /UNAVAILABE_FILE_SYSTEM
```

I think in Python you could wrap `os.stat` in a function with a timeout.

**Comment by toonn** [comment]
I think it'd be better not to stat so eagerly in the first place. I rarely navigate up to the / directory so it's a waste of time. We stat everything when navigating down, why not do the same when navigating up? All we need is a file object to select, that file object doesn't have to be completely initialized.

**Comment by MaxGyver83** [comment]
That sounds reasonable.
